gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/AbstractRegionMap.java

```diff
@@ -847,8 +849,12 @@ public void copyRecoveredEntries(RegionMap rm, boolean entriesIncompatible) {
             VersionTag tag = oldRe.getVersionStamp().asVersionTag();
             tombstones.put(tag, oldRe);
           }
-          _getOwner().updateSizeOnCreate(key,
-              _getOwner().calculateRegionEntryValueSize(oldRe));
+          // only for incrementing count while size is updated below
+          owner.updateSizeOnCreate(key, 0);
+          if (!oldRe.isOffHeap()) {
```

Comment:
Just a doubt. Is there a possibility that since accounting has moved to _setValue then in case of changeOwner where just map is borrowed from PlaceHolderDiskRegion there can be a scope of double accounting.

Reply:
@kneeraj For PlaceHolderDiskRegion the updateMemoryStats is empty so there will be no accounting done then. I have cleaned up this a bit to move the accounting inside AbstractRegionEntry.setOwner itself -- please see https://github.com/SnappyDataInc/snappy-store/pull/316/commits/dc8c5d0bd952f89cee7ae8a0a53f4791585114be